app/src/main/java/com/duckduckgo/app/trackerdetection/TrackerDetector.kt

```diff
@@ -36,6 +39,25 @@ class TrackerDetector @Inject constructor() {
     }
 
     fun shouldBlock(url: String, documentUrl: String, resourceType: ResourceType): Boolean {
-        return clients.any { it.matches(url, documentUrl, resourceType) }
+
+        if (firstParty(url, documentUrl)) {
+            Timber.v("$url is a first party url")
+            return false
+        }
+
+        val matches = clients.any { it.matches(url, documentUrl, resourceType) }
+        val matchText = if (matches) "WAS" else "was not"
```

Comment:
This line won't get stripped by ProGuard (assuming we plan to do that) and it seems to be only for the purpose of logging.

Reply:
Can you explain why this is a problem? Are you worried about efficiency? Security? don't see an issue with a convenience string containing the words "was" or "was not" existing here.

Reply:
Just that it will become dead code eventually and I personally feel a better habit would be to avoid that, but if you're not bothered it's not that big a deal.

Reply:
I would use a ternary if they were available in kotlin... but sadly not. Do you know if this gets optimised away at the bytecode level - it would in java just I can't find anything on android. Please share if you have a reference. More curious than anything, I need the variable so as you're not adamant that it needs to go I'll keep it either way.

Reply:
Do you know what, ProGuard probably strips dead code if the compiler doesn't optimise it out.